pkg/gardener/shoot/hyperscaler/aws/zones.go

```diff
@@ -28,46 +33,115 @@ cidr: 10.250.0.0/16
     public: 10.250.160.0/20
     internal: 10.250.176.0/20
 */
-func generateAWSZones(workerCidr string, zoneNames []string) []v1alpha1.Zone {
+
+func generateAWSZones(workerCidr string, zoneNames []string) ([]v1alpha1.Zone, error) {
+	numZones := len(zoneNames)
+	if numZones < minNumberOfZones || numZones > maxNumberOfZones {
+		return nil, errors.New("Number of networking zones must be between 1 and 8")
+	}
+
 	var zones []v1alpha1.Zone
 
-	cidr, _ := netip.ParsePrefix(workerCidr)
-	workerPrefixLength := cidr.Bits() + workersBits
-	workerPrefix, _ := cidr.Addr().Prefix(workerPrefixLength)
+	cidr, err := netip.ParsePrefix(workerCidr)
+	if err != nil {
+		return zones, errors.Wrap(err, "failed to parse worker CIDR")
+	}
+
+	orgWorkerPrefixLength := cidr.Bits() + workersBits
+	workerPrefix, err := cidr.Addr().Prefix(orgWorkerPrefixLength)
+	if err != nil {
+		return zones, errors.Wrap(err, "failed to get worker prefix")
+	}
 
 	// delta - it is the difference between "public" and "internal" CIDRs, for example:
 	//    WorkerCidr:   "10.250.0.0/19",
 	//    PublicCidr:   "10.250.32.0/20",
 	//    InternalCidr: "10.250.48.0/20",
-	// 4 * delta  - difference between two worker (zone) CIDRs
+	// 4 * delta  - difference between two worker (zone) CIDRs 4096 hosts
+	// small_delta and smallPrefixLength are used for subnets created for last 5 (from 4th to 8th) zones
+	// small_delta  - difference between two subnetworks CIDRs for last 5 (from 4th to 8th) zones 1024 hosts
+	//    WorkerCidr:   "10.250.192.0/22",
+	//    PublicCidr:   "10.250.196.0/22",
+	//    InternalCidr: "10.250.200.0/22",
+
+	smallPrefixLength := orgWorkerPrefixLength + 3
```

Comment:
Could you extract this `3` to a const that describes what is this number? Is it the same as `workersBits` ?